examples/MidiKeyboard/KeyboardWidget.hpp

```diff
@@ -0,0 +1,486 @@
+    {
+        fSVGs[kWhiteKeyResourceIndex].loadFromMemory(MidiKeyboardResources::white_keyData,
+                                                     MidiKeyboardResources::white_keyDataSize,
+                                                     1.0f);
+
+        fSVGs[kWhiteKeyPressedResourceIndex].loadFromMemory(MidiKeyboardResources::white_key_pressedData,
+                                                            MidiKeyboardResources::white_key_pressedDataSize,
+                                                            1.0f);
+
+        fSVGs[kBlackKeyResourceIndex].loadFromMemory(MidiKeyboardResources::black_keyData,
+                                                     MidiKeyboardResources::black_keyDataSize,
+                                                     1.0f);
+
+        fSVGs[kBlackKeyPressedResourceIndex].loadFromMemory(MidiKeyboardResources::black_key_pressedData,
+                                                    MidiKeyboardResources::black_key_pressedDataSize,
+                                                    1.0f);
+
+        for (int i = 0; i < kResourcesCount; ++i)
+        {
+            fImages[i].loadFromSVG(fSVGs[i]);
+        }
+
+        const int whiteKeysTotalSpacing = kWhiteKeySpacing * kWhiteKeysCount;
+
+        const uint width = fImages[kWhiteKeyResourceIndex].getWidth() * kWhiteKeysCount + whiteKeysTotalSpacing;
+        const uint height = fImages[kWhiteKeyResourceIndex].getHeight();
+
+        setSize(width, height);
+
+        setupKeyLookupTable();
+        setKeyImages();
+        positionKeys();
+    }
+
+    /**
+       Set the 'pressed' state of a key in the keyboard.
+     */
+    void setKeyPressed(const uint keyIndex, const bool pressed, const bool sendCallback = false)
+    {
+        DISTRHO_SAFE_ASSERT_RETURN(keyIndex < kKeyCount, )
+
+        PianoKey& key = *fKeysLookup[keyIndex];
+
+        if (key.isPressed() == pressed)
+            return;
+
+        key.setPressed(pressed);
+
+        if (fCallback != nullptr && sendCallback)
+        {
+            if (pressed)
+            {
+                fCallback->keyboardKeyPressed(keyIndex);
+            }
+            else
+            {
+                fCallback->keyboardKeyReleased(keyIndex);
+            }
+        }
+
+        repaint();
+    }
+
+    void setCallback(Callback* callback)
+    {
+        fCallback = callback;
+    }
+
+    /**
+       Draw the piano keys.
+     */
+    void onDisplay() override
+    {
+        // Draw the white keys.
+        for (int i = 0; i < kWhiteKeysCount; ++i)
+        {
+            fWhiteKeys[i].draw();
+        }
+
+        // Draw the black keys, on top of the white keys.
+        for (int i = 0; i < kBlackKeysCount; ++i)
+        {
+            fBlackKeys[i].draw();
+        }
+    }
+
+    /**
+       Get the key that is under the specified point.
+       Return nullptr if the point is not hovering any key.
+     */
+    PianoKey* tryGetHoveredKey(const Point<int>& point)
```

Comment:
This could easily be optimized, but this approach has the benefit of being simple I guess...